cmd/agent/main.go

```diff
@@ -5,16 +5,44 @@
 
 //go:build !windows
 
+// Main package for the agent binary
 package main
 
 import (
+	"fmt"
 	"os"
+	"path"
+	"strings"
 
 	"github.com/DataDog/datadog-agent/cmd/agent/command"
 	"github.com/DataDog/datadog-agent/cmd/agent/subcommands"
 	"github.com/DataDog/datadog-agent/cmd/internal/runcmd"
+	"github.com/spf13/cobra"
 )
 
+func init() {
+	registerAgent(func() *cobra.Command {
+		return command.MakeCommand(subcommands.AgentSubcommands())
+	}, "agent", "datadog-agent")
+}
+
+var agents = map[string]func() *cobra.Command{}
+
+func registerAgent(getCommand func() *cobra.Command, names ...string) {
```

Comment:
nit: I find it more readable to have the names before the `getCommand` function as the signature of the `registerAgent` function. I understand that we did this way to be able to use variadic arguments.

What do you think of having the function definition be: `func registerAgent(names []string, getCommand func() *cobra.Command)`

Reply:
Surely works for me :+1: